bgpd/bgp_route.c

```diff
@@ -2983,8 +2990,8 @@ int bgp_update(struct peer *peer, struct prefix *p, uint32_t addpath_id,
 	bgp = peer->bgp;
 	rn = bgp_afi_node_get(bgp->rib[afi][safi], afi, safi, p, prd);
 	/* TODO: Check to see if we can get rid of "is_valid_label" */
-	if (afi == AFI_L2VPN && safi == SAFI_EVPN)
-		has_valid_label = (num_labels > 0) ? 1 : 0;
+	if (is_evpn_tunnel_type_vxlan(afi, safi, attr->encap_tunneltype))
+		has_valid_label = 1;
```

Comment:
I don't undertsnad why this special case is needed.  Is label invalid in this case and if not, why?